data/transactions/logic/eval.go

```diff
@@ -547,6 +547,19 @@ func (pe PanicError) Error() string {
 var errLogicSigNotSupported = errors.New("LogicSig not supported")
 var errTooManyArgs = errors.New("LogicSig has too many arguments")
 
+// ClearStateBudgetError allows evaluation to signal that the caller should
+// reject the transaction.  Normally, an error in evaluation would not cause a
+// ClearState txn to fail. However, callers fail a txn for ClearStateBudgetError
+// because the transaction has not provided enough budget to let ClearState do
+// its job.
+type ClearStateBudgetError struct {
+	offered int
+}
+
+func (e ClearStateBudgetError) Error() string {
+	return fmt.Sprintf("ClearState execution with only %d", e.offered)
```

Comment:
nit: can the errar message refer to `PooledApplicationBudget` or something like that?

Reply:
Changed:
`return fmt.Sprintf("Attempted ClearState execution with low OpcodeBudget %d", e.offered)`
That should call to mind the `global OpcodeBudget` which is the problem.